general/tic_toc.cpp

```diff
@@ -438,10 +441,7 @@ double StopWatch::SystTime()
    return M->SystTime();
 }
 
-StopWatch::~StopWatch()
-{
-   delete M;
-}
+StopWatch::~StopWatch() = default;
```

Comment:
I think it will be better to move the `= default` to the header and remove this line.